Formula/dotnet.rb

```diff
@@ -0,0 +1,92 @@
+class Dotnet < Formula
+  desc ".Net Core"
+  homepage "https://github.com/dotnet/cli/releases"
+  url "https://github.com/dotnet/cli/archive/v1.0.0-preview2.tar.gz"
+  version "1"
+  sha256 "a0708ed3eedb10e95dcea96c62f85be1c38b91f2be9afa89d8fc655b7e0c4706"
+
+  patch :DATA
+  head "https://github.com/dotnet/cli.git"
+
+  depends_on "cmake" => :build
+  depends_on "openssl"
+
+  def install
+    system "./build.sh", "--targets", "Prepare,Compile"
+
+    stage2 = buildpath/"artifacts/osx.10.11-x64/stage2"
+    # TODO: how do I tell homebrew that the folders are under stage2?
+
+    # Relink openssl to the homebrew version
+    # Would be smoother to tell clang to use this openssl prefix, but alas...
+    openssl_path = Formula["openssl"].opt_prefix
+    system 'sudo', %W|install_name_tool
+                      -add_rpath
+                      #{openssl_path}/lib
+                      #{prefix}/shared/Microsoft.NETCore.App/1.0.0/System.Security.Cryptography.Native.dylib|
+
+    # Have a binary
+    bin.install stage2/"dotnet"
+  end
+
+  test do
+    # TODO: still to be done
+
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! It's enough to just replace
+    # "false" with the main program this formula installs, but it'd be nice if you
+    # were more thorough. Run the test with `brew test cli`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    system "artifacts/darwin-x64/stage2/dotnet", "version"
```

Comment:
The test runs against the installed binaries and not from `buildroot`; you probably want `system bin/"dotnet", ...`.